framework/OutStreams/Factory.py

```diff
@@ -22,29 +22,22 @@
 
 ################################################################################
 from utils import utils
-from .OutStreamManager import OutStreamManager
-from .OutStreamPlot import OutStreamPlot
-from .OutStreamPrint import OutStreamPrint
-## [ Add new class here ]
-################################################################################
-## Alternatively, to fully automate this file:
-# from OutStreamManagers import *
-################################################################################
 
-"""
-  Interface Dictionary (factory) (private)
-"""
-# This machinery will automatically populate the "knownTypes" given the
-# imports defined above.
-__base = 'OutStreamManager'
-__interFaceDict = {}
+from .OutStreamBase import OutStreamBase
+
+from .FilePrint import FilePrint
+from .GeneralPlot import GeneralPlot
+# from .DataMining import DataMining
+# from .VDCComparison import VDCComparison
 
-for classObj in utils.getAllSubclasses(eval(__base)):
-  ## As long as these subclasses follow the pattern of starting with OutStream
-  ## this will appropriately key them to a more user-friendly name without the
-  ## need for them to redudantly prepend "X" as "OutStreamX"
-  key = classObj.__name__.replace('OutStream','')
-  __interFaceDict[key] = classObj
+# Interface Dictionary (factory) (private)
+__base = 'OutStreamBase'
+__interFaceDict = {
+  'Print': FilePrint,
+  'Plot': GeneralPlot,
+  # 'DataMining': DataMining,
+  # 'VDCComparison': VDCComparison,
```

Comment:
In my opinion the specializations of the Plots/Prints should be under Plot/Print (similarly to External Model/type or Code, CodeInterfaces)

Reply:
placeholder for discussion

Reply:
I'm noticing immediately that there would be a big benefit for a base class for `Plot`, although I think the Entity approach is cleaner. Maybe in the future:
```xml
<OutStreams>
    <Print>
        <MyCustomPrint>
          <... options ...>
        </MyCustomPrint>
    </Print>
</OutStreams>
```
although there's a couple things I don't love about this:
- It forces an OutStream to be only a Print or a Plot (I can imagine cases where both make sense)
- It is regressing towards the <subType> approach in the <Step> listing `class=OutStream, type=Print, subType=MyPrint` instead of `class=OutStream type=MyPrint`
- We have to modify all existing RAVEN inputs if we add a subspot under <Print> for <FilePrint>, or some fancy checking to see if we're dealing with a special type or the generic type.

The other two options I see are the way I hinted at here (`OutStream.MyCustomPrint`) or using the `subType` approach from the ROMs, that I think we don't like much.

Definitely a good discussion to have when we start designing the first new entries.